Dependencies/napi/include/napi/env.h

```diff
@@ -2,11 +2,19 @@
 
 #include "napi.h"
 
+typedef const struct OpaqueJSContext* JSContextRef;
+typedef struct OpaqueJSContext* JSGlobalContextRef;
+
+
 namespace Napi
 {
     template<typename ...Ts> Napi::Env Attach(Ts... args);
 
     void Detach(Napi::Env);
 
     Napi::Value Eval(Napi::Env env, const char* source, const char* sourceUrl);
-}
\ No newline at end of file
+
+#ifdef __APPLE__
+    JSGlobalContextRef GetJavaScriptCoreGlobalContext(Napi::Env env);
```

Comment:
This seems really weird. You're using Objective-C, right? Can you get the context [this way](https://developer.apple.com/documentation/javascriptcore/jscontext/1451545-currentcontext) instead?

Reply:
[JSContext currentContext] returns nil. At that point, we are not executing JS. That method would return a valid context inside a napi function. Not at init time.

Reply:
Then can't we just call it from inside JS? Since we have the `Env` and should already be on the right thread, just create a JS function that calls straight back into native, then call that. That'll get JSC on the stack, which should then cause that function to return a legitimate value. Would that work?

Reply:
I really don't think we can put platform- and engine-specific concepts into `env.h` itself.

Reply:
> Then can't we just call it from inside JS?
Do you mean registering the JSC class from a napi function that's being called?

Reply:
There doesn't appear to be an easy way to get the JS context using `currentContext`. The next best thing would be to expose cleanly from JsRuntime. Exposing it from Napi::Env is probably not a good idea. Let's chat about it offline.